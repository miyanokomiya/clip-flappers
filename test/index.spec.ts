import Target from '../src/index'
import { image200x100 } from './assets'

function expectToShow($el: HTMLElement | SVGElement) {
  expect($el.style.display).toBe('')
}
function expectToHide($el: HTMLElement | SVGElement) {
  expect($el.style.display).toBe('none')
}

describe('index', () => {
  describe('snapshot', () => {
    it('default', () => {
      const $el = document.createElement('div')
      const app = new Target($el) as any

      expectToShow(app.$el.querySelector('[data-key="clip-f_drop-button"]'))
      expectToHide(app.$el.querySelector('[data-key="clip-f_delete-button"]'))
      expectToHide(app.$svg)
      expect(app.$el).toMatchSnapshot()
    })
    it('image loaded', async () => {
      const $el = document.createElement('div')
      const app = new Target($el) as any
      app.base64 = image200x100
      await app.updateImage()

      expectToHide(app.$el.querySelector('[data-key="clip-f_drop-button"]'))
      expectToShow(app.$el.querySelector('[data-key="clip-f_delete-button"]'))
      expectToShow(app.$svg)
      expect(app.$el).toMatchSnapshot()
    })
  })

  describe('reset', () => {
    it('back to default view', async () => {
      const $el = document.createElement('div')
      const app = new Target($el) as any
      app.base64 = image200x100
      await app.updateImage()
      app.$el.querySelector('[data-key="clip-f_delete-button"]').click()
      expectToShow(app.$el.querySelector('[data-key="clip-f_drop-button"]'))
      expectToHide(app.$el.querySelector('[data-key="clip-f_delete-button"]'))
      expectToHide(app.$svg)
    })
  })

  describe('dispolse', () => {
    it('remove all DOM', async () => {
      const $el = document.createElement('div')
      const app = new Target($el) as any
      app.base64 = image200x100
      await app.updateImage()
      app.dispose()

      expect(app.$el).toBe(null)
      expect(app.$svg).toBe(null)
      expect(app.$clipRect).toBe(null)
      expect($el.innerHTML).toBe('')
    })
  })
})
