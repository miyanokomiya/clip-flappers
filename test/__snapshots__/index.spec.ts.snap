// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index snapshot default 1`] = `
<div>
  <div>
    <input
      accept="image/*"
      style="display: none;"
      type="file"
    />
    <div
      style="padding:8px;border:1px solid #000;backgroundColor:#ccc;overflow:hidden;position:relative;width:124px;height:124px;"
    >
      <svg
        style="overflow: visible; width: 100%; height: 100%; display: none;"
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      />
      <button
        data-key="clip-f_drop-button"
        style="padding:0;margin:0;border:none;background-color:transparent;width:100%;height:100%;cursor:pointer;"
      >
        <svg
          style="overflow:visible;width:100%;height:100%;"
          viewBox="0 0 100 100"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            transform="translate(25,25)"
          >
            <g
              fill="#aaa"
            >
              <path
                d="M0,45 L0,40 L15,25 L20,30 L30,20 L50,40 L50,45z"
              />
              <circle
                cx="10"
                cy="15"
                r="4"
              />
            </g>
            <rect
              fill="none"
              height="50"
              rx="4"
              ry="4"
              stroke="#aaa"
              stroke-width="2"
              width="50"
            />
          </g>
        </svg>
      </button>
      <button
        data-key="clip-f_delete-button"
        style="padding: 0px; margin: 0px; background-color: transparent; position: absolute; top: 4px; right: 4px; width: 20px; height: 20px; border-radius: 100%; cursor: pointer; display: none;"
      >
        <svg
          style="overflow:visible;width:100%;height:100%;"
          viewBox="0 0 100 100"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M24,24 L76,76 M24,76 L76,24"
              fill="none"
              stroke="#aaa"
              stroke-width="12"
            />
            <circle
              cx="50"
              cy="50"
              fill="none"
              r="50"
              stroke="#aaa"
              stroke-width="10"
            />
          </g>
        </svg>
      </button>
    </div>
  </div>
</div>
`;

exports[`index snapshot image loaded 1`] = `
<div>
  <div>
    <input
      accept="image/*"
      style="display: none;"
      type="file"
    />
    <div
      style="padding:8px;border:1px solid #000;backgroundColor:#ccc;overflow:hidden;position:relative;width:124px;height:124px;"
    >
      <svg
        style="overflow: visible; width: 100%; height: 100%;"
        viewBox="0 -50 200 200"
        xmlns="http://www.w3.org/2000/svg"
      >
        <image
          height="100"
          href="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAMgAAABkBAMAAAAxqGI4AAAAG1BMVEUAAAD///8/Pz+/v7/f399/f3+fn58fHx9fX18kPjWPAAAACXBIWXMAAA7EAAAOxAGVKw4bAAABi0lEQVRoge2UMU/DMBCFjySNM8LA7tJIrBWtEGMJlGbsgMQKUtlDWzFbIj+cs88hoaVSYzq+T4ot6z35xbmLiQAAAAAAAAAAAACO4krbUV25xVofcK27rr5E08n9kqfHyZxX+cN0uet40z+Cd/Um1aSeeBNNm1eKClK3O4bNikO8IK4whhWVPJWUGHJPBzVPNDWCuMJIdDbj6YtqHrPxLy03NkQE7wojNqmtxAe986jOeRjwxvRptReyISJ4VxhnVWx4Sqo7u1qQqwEp/2FsiAjeFRZSu40o1oVd3diBS7w1bYgI3hUWUsrrxcYeQt46WzQHcSEieFdQxmC8H0J13Wx2mpC62v9cNLhs5JN8rmjhX69TeKLtqinwSQrPNabdFubWyopOyH9bOHMtOyP7m+XU/IzcWrVuQ0TwrgCe3fFtKxWUGNnT/SNR0YZ4QVz9SaUEuXEXZOkvyJQ3pVEb4gVx9Wd4wSztjW8v8dEfV707mgjeFYy6dtP6UO+I4F0AAAAAAAAAAAAAR/ENV6VF0qC6xGQAAAAASUVORK5CYII="
          width="200"
          x="0"
          y="0"
        />
        <g
          transform="translate(0, -50)"
        >
          <rect
            fill="none"
            height="200"
            stroke="red"
            stroke-width="6.451612903225806"
            width="200"
            x="0"
            y="0"
          />
          <g>
            <circle
              cx="0"
              cy="0"
              fill="red"
              r="12.903225806451612"
              stroke="none"
            />
          </g>
          <g
            transform="translate(200, 200)"
          >
            <circle
              cx="0"
              cy="0"
              fill="red"
              r="12.903225806451612"
              stroke="none"
            />
          </g>
        </g>
      </svg>
      <button
        data-key="clip-f_drop-button"
        style="padding: 0px; margin: 0px; background-color: transparent; width: 100%; height: 100%; cursor: pointer; display: none;"
      >
        <svg
          style="overflow:visible;width:100%;height:100%;"
          viewBox="0 0 100 100"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            transform="translate(25,25)"
          >
            <g
              fill="#aaa"
            >
              <path
                d="M0,45 L0,40 L15,25 L20,30 L30,20 L50,40 L50,45z"
              />
              <circle
                cx="10"
                cy="15"
                r="4"
              />
            </g>
            <rect
              fill="none"
              height="50"
              rx="4"
              ry="4"
              stroke="#aaa"
              stroke-width="2"
              width="50"
            />
          </g>
        </svg>
      </button>
      <button
        data-key="clip-f_delete-button"
        style="padding: 0px; margin: 0px; background-color: transparent; position: absolute; top: 4px; right: 4px; width: 20px; height: 20px; border-radius: 100%; cursor: pointer;"
      >
        <svg
          style="overflow:visible;width:100%;height:100%;"
          viewBox="0 0 100 100"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M24,24 L76,76 M24,76 L76,24"
              fill="none"
              stroke="#aaa"
              stroke-width="12"
            />
            <circle
              cx="50"
              cy="50"
              fill="none"
              r="50"
              stroke="#aaa"
              stroke-width="10"
            />
          </g>
        </svg>
      </button>
    </div>
  </div>
</div>
`;
